crates/router/src/connector/bluesnap/transformers.rs

```diff
@@ -116,10 +127,82 @@ impl TryFrom<&types::PaymentsAuthorizeRouterData> for BluesnapPaymentsRequest {
             payment_method,
             currency: item.request.currency,
             card_transaction_type: auth_mode,
+            three_d_secure: None,
+        })
+    }
+}
+
+impl TryFrom<&types::PaymentsCompleteAuthorizeRouterData> for BluesnapPaymentsRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsCompleteAuthorizeRouterData) -> Result<Self, Self::Error> {
+        let redirection_response: BluesnapRedirectionResponse = item
+            .request
+            .payload
+            .clone()
+            .ok_or(errors::ConnectorError::MissingRequiredField {
+                field_name: "bluesnap_redirection_response",
+            })?
+            .parse_value("BluesnapRedirectionResponse")
+            .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
+
+        let redirection_result: BluesnapThreeDsResult = redirection_response
+            .authentication_response
+            .parse_struct("BluesnapThreeDsResult")
+            .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
+
+        let auth_mode = match item.request.capture_method {
+            Some(enums::CaptureMethod::Manual) => BluesnapTxnType::AuthOnly,
+            _ => BluesnapTxnType::AuthCapture,
+        };
+        let payment_method = if let Some(api::PaymentMethodData::Card(ccard)) =
+            item.request.payment_method_data.clone()
+        {
+            PaymentMethodDetails::CreditCard(Card {
+                card_number: ccard.card_number,
+                expiration_month: ccard.card_exp_month.clone(),
+                expiration_year: ccard.card_exp_year.clone(),
+                security_code: ccard.card_cvc,
+            })
+        } else {
+            Err(errors::ConnectorError::MissingRequiredField {
+                field_name: "PaymentMethod",
+            })?
+        };
+        Ok(Self {
+            amount: utils::to_currency_base_unit(item.request.amount, item.request.currency)?,
+            payment_method,
+            currency: item.request.currency,
+            card_transaction_type: auth_mode,
+            three_d_secure: Some(BluesnapThreeDSecureInfo {
+                three_d_secure_reference_id: redirection_result
+                    .three_d_secure
+                    .ok_or(errors::ConnectorError::MissingRequiredField {
```

Comment:
add a new error for `MissingPayloadField{fieldname}`, MissingRequiredField is reserved for customer missing an required field. this is not customer input.